src/app/core/entity-components/entity-utils/dynamic-form-components/edit-single-entity/edit-single-entity.component.ts

```diff
@@ -52,7 +63,40 @@ export class EditSingleEntityComponent
       (entity) => entity.getId() === this.formControl.value
     );
     if (selectedEntity) {
+      this.selectedEntity = selectedEntity;
+      this.editingSelectedEntity = false;
       this.entityNameFormControl.setValue(selectedEntity.toString());
+    } else {
+      this.entityNameFormControl.setValue("");
     }
   }
+
+  select(event: MatAutocompleteSelectedEvent) {
+    const entity = this.entities.find(
+      (e) => e.toString().toLowerCase() === event.option.value.toLowerCase()
+    );
+    if (entity) {
+      this.selectedEntity = entity;
+      this.editingSelectedEntity = false;
+      this.formControl.setValue(entity.getId());
+    }
+  }
+
+  submit() {
+    const currentEntity = this.entityNameFormControl.value;
+    const entity = this.entities.find((e) => e.toString() === currentEntity);
+    if (entity) {
+      this.selectedEntity = entity;
+      this.editingSelectedEntity = false;
+      this.formControl.setValue(entity.getId());
+    }
+  }
+
+  editSelectedEntity() {
+    this.editingSelectedEntity = true;
+    this.formControl.setValue(null);
```

Comment:
Why ist this set to `null`?

Reply:
There could be the case where the user edits the selected entity, enters an invalid entity, and exits the form field. If the value wasn't set to `null`, the wrong value would still be valid.